src/wrappers/themis/wasm/test/test.js

```diff
@@ -36,14 +36,14 @@ function measureTime(thunk) {
 
 describe('wasm-themis', function() {
     let generallyInvalidArguments = [
-        null, undefined, 'string',
+        null, 'string',
         new Int16Array([1, 2, 3]), [4, 5, 6],
         () => new Uint8Array([27, 18, 28, 18, 28]),
         { value: [3, 14, 15, 92, 6] }
     ]
     describe('initialization', function() {
         it('resolves "initialized" promise', function(done) {
-            themis.initialized.then(function() {
+            themis.initialize().then(function() {
```

Comment:
>   @maxammann, is my reasoning correct so far?

Perfectly correct!

> 
>   However, I believe it's not be nice to snatch the rag from existing users and remove `initialized` entirely. Would it be possible for `initialized` and `initialize()` to coexist for a while?
> 

I will wee whether that is possible. I can not start loading themis as soon as this library is importet as you may want to configure the path to the wasm binary (required for web and webpack).

> * Changing the base type of private/public keys.

Supporting es5 is not possible with that extends syntax unfortunately. I had to support and target es5 such that the integration CI passes. So I think we should support it. The reason why it worked before is that without a compilation it somehow worked as node supports a mix between es5 and es6. Some browsers do not and only support es5. Therefore this kind of fixed bugs for users of older browsers.


BUT: We could drop support for ES5 actually because every browser which supports webassambly should already support ES6. This is also the reason why we do not need to support IE11. Webassambly does not work there in the first place.

Reply:
> Perfectly correct!

Whew 😌 Thanks for confirming.

> I will wee whether that is possible. I can not start loading themis as soon as this library is importet as you may want to configure the path to the wasm binary (required for web and webpack).

We have not really considered – and tested – the web usage. Knows users of WasmThemis are all on Node.js.

I remember doing some manual tests and that it kinda worked. But I'm no web developer so I believe no one so far has tested WasmThemis with proper web apps. Hence, I wouldn't be really surprised that it does not work out of the box. If something is necessary for proper web support – go ahead, you have my blessing to add whatever is needed 👍

It would be nice to keep the `initialized` promise around – I'm sorry, I don't have any ideas right now – but it it's not possible, then, well, WasmThemis users are getting a flag day. This one should not be too painful to work around.

> BUT: We could drop support for ES5 actually because every browser which supports webassambly should already support ES6.

Uh-huh... I faintly remember this reasoning behind setting the minimal supported version to ES6 and using all those fancy things.

Personally, I'm fine with ES5 support being added, but probably not at an expense of backwards-incompatible changes to the API which are caused by ES5 lack of features. If there was a way to give ES5 users one code and ES6 gets a different one – that would be great, but browsers don't work like that: it's easier to serve the same version to everyone without trying to guess what their browser might or might not support. So the approach would be something like "use some magic transpiler to convert ES6 code into ES5-compatible code".